lib/api.js

```diff
@@ -104,22 +82,30 @@ module.exports.methods = {
                     sender.send(options, cb, fail);
                 }, delay);
             };
-        debug.write('Sending logs: batch size: ' + messages.length + ' messages');
+        debug.write('[Log] Sending logs: batch size: ' + messages.length + ' messages');
         sender.send(options, cb, shutdown ? null : fail);
     },
     /*
     *** posting logs synchronously in case if server is about to close ***
     */
     postLogsSync: function postLogsSync(messages) {
-        if (CONFIG.TRANSPORT === 'agent_socket') {
-            module.exports.methods.sendToSocket(messages); // send to unix socket
-        } else {
-            var options = {
-                url: CONFIG.PROTOCOL + '://' + CONFIG.HOST + CONFIG.LOG_SAVE_PATH,
-                headers : helpers.getHeaders(),
-                data: helpers.getPostBody(messages)
-            };
-            sender.sendSync(options);
+        var cb = function (response) {
+            debug.write('postLogsSync response: ' + response)
+        }
+        switch(CONFIG.TRANSPORT) {
+            case 'agent_socket':
+                module.exports.methods.sendToSocket(messages, cb); // send to unix socket
+                break;
+            case 'agent_http':
+                module.exports.methods.sendToHttpClient(messages, cb); // send to http
+                break;
+            default:
```

Comment:
suggest changing from default to a case statement for 'log' - have the default write a warn message for an unsupported transport.

Reply:
I've updated the case statement changing from default to log.